### DISCUSSION TITLE
Reduce generics params

### DISCUSSION THREAD

**Discussion by mikasoukhov** [dotnet/csharplang#5430] [Language Ideas]
Declaration

```
public interface IEntityService<T> {}

public interface IFileService<File>
{
}

public class MyClass<TEntity, TSerivce> where TService : IEntityService<TEntity>
{
}
```

Usage (now)

`var obj = new MyClass<File, IFileService>();`

Usage (can be)

`var obj = new MyClass<_, IFileService>();`

**Comment by jnm2** [comment]
~~https://github.com/dotnet/csharplang/issues/1349 discusses options that cover this use case (with varying syntax).~~

**Reply by YairHalberstadt** [reply]
I don't think it would, since that doesn't cover constructors

**Reply by jnm2** [reply]
Oh oops, thank you.

**Reply by jnm2** [reply]
I thought there was an existing issue/discussion on omitting generic type arguments in any call (method or constructor), using either `_` or empty space.

**Reply by YairHalberstadt** [reply]
Constructor type inference is discussed at https://github.com/dotnet/csharplang/discussions/281, but not with respect to partial type inference.
